Lib9c/Action/HackAndSlash.cs

```diff
@@ -274,81 +230,94 @@ public override IAccountStateDelta Execute(IActionContext context)
             Log.Verbose("{AddressesHex}HAS Get skillState : {Elapsed}", addressesHex, sw.Elapsed);
 
             sw.Restart();
-            var simulator = new StageSimulator(
-                random,
-                avatarState,
-                foods,
-                skillsOnWaveStart,
-                worldId,
-                stageId,
-                sheets.GetStageSimulatorSheets(),
-                sheets.GetSheet<CostumeStatSheet>(),
-                StageSimulator.ConstructorVersionV100080);
-
-            sw.Stop();
-            Log.Verbose("{AddressesHex}HAS Initialize Simulator: {Elapsed}", addressesHex, sw.Elapsed);
-
+            var worldSheet = sheets.GetSheet<WorldSheet>();
+            var worldUnlockSheet = sheets.GetSheet<WorldUnlockSheet>();
+            var crystalStageBuffSheet = sheets.GetSheet<CrystalStageBuffGachaSheet>();
             sw.Restart();
-            simulator.Simulate(1);
-            sw.Stop();
-            Log.Verbose("{AddressesHex}HAS Simulator.Simulate(): {Elapsed}", addressesHex, sw.Elapsed);
-
-            Log.Verbose(
-                "{AddressesHex}Execute HackAndSlash({AvatarAddress}); worldId: {WorldId}, stageId: {StageId}, result: {Result}, " +
-                "clearWave: {ClearWave}, totalWave: {TotalWave}",
-                addressesHex,
-                avatarAddress,
-                worldId,
-                stageId,
-                simulator.Log.result,
-                simulator.Log.clearedWaveNumber,
-                simulator.Log.waveCount
-            );
-
-            if (simulator.Log.IsClear)
+            // if PlayCount > 1, it is Multi-HAS.
+            for (var i = 0; i < PlayCount; i++)
             {
                 sw.Restart();
-                simulator.Player.worldInformation.ClearStage(
-                    worldId,
-                    stageId,
-                    blockIndex,
-                    worldSheet,
-                    sheets.GetSheet<WorldUnlockSheet>()
-                );
+                // First simulating will use Foods and Random Skills.
+                // Remainder simulating will not use Foods.
+                var simulator = new StageSimulator(
+                    new HackAndSlashRandom(random.Next()),
```

Comment:
Whats the different between using has `HackAndSlashRandom` and `random`?

Reply:
I expected a problem with reusing random objects, but I was wrong.
I will remove `HackAndSlashRandom`. 😅